### PR TITLE
enable dataset functionality in nextflow schema build

### PR DESCRIPTION
**Pull request by dorien-er**

### CODE REVIEW

File: pro/build-nextflow-schemas/action.yml [hunk 1]
```diff
@@ -19,6 +19,10 @@ inputs:
       Release of Viash tools to use.
     required: false
     default: 'latest'
+  enable_dataset_input:
```
Comment (DriesSchaumont) [inline pro/build-nextflow-schemas/action.yml:22]: Could you add the input type (and perhaps also to the other arguments)? Here is an example: https://github.com/openpipelines-bio/openpipeline/blob/e1895f1c70f820796581d5889bdd948aa18cb286/.github/workflows/main-build.yml#L9C7-L9C22

File: pro/build-nextflow-schemas/action.yml [hunk 2]
```diff
@@ -106,8 +110,12 @@ runs:
           outputs+=(--output "$component_outputs_schema")
         fi
 
-
-        echo "Running viash_tools/target/docker/nextflow/generate_schema/generate_schema ${inputs[@]} ${outputs[@]}"
-        viash_tools/target/docker/nextflow/generate_schema/generate_schema "${inputs[@]}" "${outputs[@]}"
-        echo "Created $outputs_schema"
-      
\ No newline at end of file
+        if [ "${{ inputs.enable_dataset_input }}" ]; then
```
Comment (DriesSchaumont) [inline pro/build-nextflow-schemas/action.yml:113]: I don't think this will work... you might need the following suggestion. I remember that I used to have problems with this in github actions in the past.

```suggestion
        if [ "${{ inputs.enable_dataset_input }}" == "true" ]; then
```

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: pro/build-nextflow-schemas/action.yml (hunk 2 above)
Comment (dorien-er) [reply pro/build-nextflow-schemas/action.yml:113]: ran a test build in OP and it worked without the string conversion as well, but can convert it to avoid future trouble

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# viash-actions v5.0.1
+
+## New functionality
+
+* `pro/build-nextflow-schemas`: Add optional flag to allow dataset input for nf-tower.
```
Comment (DriesSchaumont) [inline CHANGELOG.md:5]: Forgot to look at this. Could you add a PR number to this line

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+# viash-actions v5.0.1
```
Comment (DriesSchaumont) [inline CHANGELOG.md:1]: I think this should be `5.1.0`

File: pro/build-nextflow-schemas/action.yml [hunk 5]
```diff
@@ -106,8 +114,12 @@ runs:
           outputs+=(--output "$component_outputs_schema")
         fi
 
-
-        echo "Running viash_tools/target/docker/nextflow/generate_schema/generate_schema ${inputs[@]} ${outputs[@]}"
-        viash_tools/target/docker/nextflow/generate_schema/generate_schema "${inputs[@]}" "${outputs[@]}"
-        echo "Created $outputs_schema"
-      
\ No newline at end of file
+        if [ "${{ inputs.enable_dataset_input }}" == "true" ]; then
```
Comment (DriesSchaumont) [inline pro/build-nextflow-schemas/action.yml:117]: I think something like this should also work
```
extra_params=()
if [ "${{ inputs.enable_dataset_input }}" == "true" ]; then
    extra_params=+=("--enable_dataset_input")
fi
echo "Running viash_tools/target/docker/nextflow/generate_schema/generate_schema ${inputs[@]} ${outputs[@]} {extra_params[@]}"
viash_tools/target/docker/nextflow/generate_schema/generate_schema "${inputs[@]}" "${outputs[@]}" "${extra_params[@]}"
```

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -127,4 +133,4 @@
 Initial release. Contains the following actions:
 
 * `setup`: Install Viash
-* `ns-list`: List components in repo
\ No newline at end of file
+* `ns-list`: List components in repo
```
Comment (DriesSchaumont) [inline CHANGELOG.md:136]: Could you undo this change?

**Review by DriesSchaumont — APPROVED** [review]
ALGTM! This can be merged once this accidental change is undone.

**Comment by DriesSchaumont** [comment]
Wohoo, thanks @dorien-er!

**Comment by dorien-er** [comment]
> Wohoo, thanks @dorien-er!

thanks you!
